### PR TITLE
ui. show dhcp lease expiration

### PR DESCRIPTION
**Pull request by federicoballarini**
Add lease expiration column on DHCP page.
Refer to: https://github.com/NethServer/dev/issues/5972

### CODE REVIEW

**Comment by federicoballarini** [comment]
Output should be something like this:

Schermata 2019-12-01 alle 15 19 32

File: ui/src/components/system/DHCP.vue [hunk 1]
```diff
@@ -92,13 +92,17 @@
...
           {{props.row.props.IpAddress}}
         </td>
         <td class="fancy">
-          <span class="pficon pficon-plugged"></span>
+          <span class="pficon pficon-plugged" style="margin-right: 5px"></span>
           {{props.row.props.MacAddress}}
         </td>
+        <td class="fancy">
+          <span v-show="props.row.props.LeaseExpiration" class="fa fa-hourglass-end" style="margin-right: 5px"></span>
+          {{props.row.props.LeaseExpiration}}
```
Comment (edospadoni) [inline ui/src/components/system/DHCP.vue]: add `dateFormat` filter here:

`{{props.row.props.LeaseExpiration | dateFormat}}`

File: ui/src/components/system/DHCP.vue [hunk 2]
```diff
@@ -988,7 +997,8 @@ export default {
                 props: {
                   Description: "",
                   IpAddress: lease.ip,
-                  MacAddress: lease.mac
+                  MacAddress: lease.mac,
+                  LeaseExpiration: new Date(lease.expire*1000).toLocaleString(context.$i18n.locale, {dateStyle: "medium", timeStyle: "medium", hour12: false, timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone}).split(",").join(" ")
```
Comment (edospadoni) [inline ui/src/components/system/DHCP.vue]: remove all Date formatting and keep only:

`LeaseExpiration: lease.expire`

File: ui/src/components/system/DHCP.vue [hunk 3]
```diff
@@ -92,13 +92,17 @@
         </td>
         <td class="fancy">{{ props.row.props.Description}}</td>
         <td class="fancy">
-          <span class="fa fa-desktop"></span>
+          <span class="fa fa-desktop" style="margin-right: 5px"></span>
```
Comment (edospadoni) [inline ui/src/components/system/DHCP.vue]: remove inline style, add an existing class or create new one

File: ui/src/components/system/DHCP.vue [hunk 4]
```diff
@@ -92,13 +92,17 @@
...
           {{props.row.props.IpAddress}}
         </td>
         <td class="fancy">
-          <span class="pficon pficon-plugged"></span>
+          <span class="pficon pficon-plugged" style="margin-right: 5px"></span>
```
Comment (edospadoni) [inline ui/src/components/system/DHCP.vue]: remove inline style, add an existing class or create new one

**Comment by federicoballarini** [comment]
Changed. It should be ok now.

**Comment by federicoballarini** [comment]
There is a problem with sorting. I think it's caused by `-` that returns `dateFormat` if no date is found.
I think we can change line 104 into

`<span v-show="props.row.props.LeaseExpiration">{{props.row.props.LeaseExpiration | dateFormat}}</span>`

What do you think? @edospadoni

**Comment by edospadoni** [comment]
> There is a problem with sorting. I think it's caused by `-` returned by `dateFormat` if no date is found.
> I think we can change line 104 into
> 
> `<span v-show="props.row.props.LeaseExpiration">{{props.row.props.LeaseExpiration | dateFormat}}</span>`
> 
> What do you think? @edospadoni

The problem is on sorting function. You have to specify field type, in this case `number`, so add:

```js
type: "number"
```

in column definition

File: ui/src/components/system/DHCP.vue [hunk 5]
```diff
@@ -682,6 +686,11 @@ export default {
           field: "props.MacAddress",
           filterable: true
         },
+        {
+          label: this.$i18n.t("dhcp.lease_expiration"),
+          field: "props.LeaseExpiration",
+          filterable: true
```
Comment (edospadoni) [inline ui/src/components/system/DHCP.vue]: add `type: "number"` here

File: ui/src/components/system/DHCP.vue [hunk 6]
```diff
@@ -92,13 +92,17 @@
...
+          <span class="fa fa-desktop margin"></span>
           {{props.row.props.IpAddress}}
         </td>
         <td class="fancy">
-          <span class="pficon pficon-plugged"></span>
+          <span class="pficon pficon-plugged margin"></span>
           {{props.row.props.MacAddress}}
         </td>
+        <td class="fancy">
+          <span v-show="props.row.props.LeaseExpiration" class="fa fa-hourglass-end margin"></span>
```
Comment (edospadoni) [inline ui/src/components/system/DHCP.vue]: remove `v-show="props.row.props.LeaseExpiration"`

File: ui/src/components/system/DHCP.vue (hunk 5 above)
Comment (federicoballarini) [reply ui/src/components/system/DHCP.vue]: Yes, you're right! Code genius :)

File: ui/src/components/system/DHCP.vue (hunk 6 above)
Comment (federicoballarini) [reply ui/src/components/system/DHCP.vue]: Why do you want to remove it? It will show the hourglasses even if there is not lease expiration.

Comment (edospadoni) [reply ui/src/components/system/DHCP.vue]: If there is no `lease expiration` a dash (`-`) is shown but the hourglass is always present to indicat the type of the field. But if you prefer to remove it no problem is only a cosmetis.

Comment (federicoballarini) [reply ui/src/components/system/DHCP.vue]: Yes, ok. We can leave it shown. I proceed to update my PR.
Thanks.

**Pull request merged**